### ISSUE TITLE
cgen: sql + time = error

### ISSUE THREAD

**Issue by enghitalo** [vlang/v#16374]
**V doctor:**
```
OS: linux, Ubuntu 22.04.1 LTS
Processor: 8 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-11390H @ 3.40GHz
CC version: cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0

getwd: /home/hitalo/Documents/daily_api_v
vmodules: /home/hitalo/.vmodules
vroot: /home/hitalo/v
vexe: /home/hitalo/v/v
vexe mtime: 2022-11-09 11:59:30
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 bf00e9b.dd5f618

Git version: git version 2.34.1
Git vroot status: weekly.2022.45-8-gdd5f6186
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg src/main.v`
```v
module main

import sqlite
import time

[table: 'taskx']
struct Task {
mut:
	id            u32    [primary; serial; sql: serial]
	frequency     string [required; sql_type: 'varchar(10)']
	deadline_dd string [required; sql_type: 'varchar(5)']
}

pub fn create_db_connection() !sqlite.DB {
	mut db := sqlite.connect('databasx.db')!
	return db
}

fn main() {

	now := time.now()
	mut db := create_db_connection() or { panic(err) }

	sql db {
		create table Task
	} or { panic('error on create table: $err') }

	result := sql db {
		select from Task where frequency == frequency
		&& deadline_dd == now.custom_format('DD')
	} or { eprintln('Error on search task: \n$err') }
	
	println(result)
}
```

**What did you expect to see?**

What did you expect to see? Return of query, in this case, []

**What did you see instead?**
```
================ V panic ================
   module: v.ast
 function: default_table_panic_handler()
  message: sym: invalid type (typ=ast.Type(0x0 = 0) idx=0). Compiler bug. This should never happen. Please report the bug using `v bug file.v`.

     file: /home/hitalo/v/vlib/v/ast/table.v:84
   v hash: dd5f618
=========================================
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/builtin/builtin.c.v:53: at panic_debug: Backtrace
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/ast/table.v:84: by v__ast__default_table_panic_handler
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/ast/table.v:90: by v__ast__Table_panic
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/ast/table.v:716: by v__ast__Table_sym
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/sql.v:340: by v__gen__c__Gen_sql_write_orm_primitive
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/sql.v:327: by v__gen__c__Gen_sql_expr_to_orm_primitive
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/sql.v:485: by v__gen__c__Gen_sql_gen_where_data
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/sql.v:667: by v__gen__c__Gen_sql_select
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/sql.v:562: by v__gen__c__Gen_sql_select_expr
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:3353: by v__gen__c__Gen_expr
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/assign.v:467: by v__gen__c__Gen_assign_stmt
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:1804: by v__gen__c__Gen_stmt
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:1725: by v__gen__c__Gen_stmts_with_tmp_var
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:1619: by v__gen__c__Gen_stmts
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/fn.v:381: by v__gen__c__Gen_gen_fn_decl
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/fn.v:109: by v__gen__c__Gen_fn_decl
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:1925: by v__gen__c__Gen_stmt
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:1725: by v__gen__c__Gen_stmts_with_tmp_var
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:1619: by v__gen__c__Gen_stmts
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:704: by v__gen__c__Gen_gen_file
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/v/gen/c/cgen.v:653: by v__gen__c__cgen_process_one_file_cb
/tmp/v_1000/../../../../../../home/hitalo/v/vlib/sync/pool/pool.v:113: by sync__pool__process_in_thread
/tmp/v_1000/vdbg.1601477665671447558.tmp.c:10941: by sync__pool__process_in_thread_thread_wrapper
7fe383c29b43 : by ???
```

**Comment by walkingdevel** [comment]
I fixed it in #17127, but there is one problem in `where frequency == frequency`. First `frequency` is a structure field, and the second is not defined above. I'll fix it too.

**Comment by walkingdevel** [comment]
In #17136 code from the issue throws an error now:
```
time_error.v:29:39: error: undefined variable: `frequency`
   27 | 
   28 |     result := sql db {
   29 |         select from Task where frequency == frequency
      |                                             ~~~~~~~~~
   30 |         && deadline_dd == now.custom_format('DD')
   31 |     } or { eprintln('Error on search task: \n$err') }
```

After adding the `frequency` variable before SQL code compiles with no errors.
